src/DistrictResources/DistrictResources.h

```diff
@@ -2,17 +2,27 @@
 #define LOCK_FREE_PROGRAMMING_DISTRICTRESOURCES_H
 
 #include <vector>
+#include <unordered_map>
+#include <memory>
 #include "../Vehicle/Vehicle.h"
+#include "../Event/Event.h"
 
 namespace LockFreeDispatch {
 
     class DistrictResources {
         private:
+            std::vector< std::unique_ptr< Vehicle > > districtVehicles;
+            std::vector< std::unique_ptr < FireStation > > districtFireStations;
+            std::unordered_map<uint32_t, std::vector<Vehicle*> > vehicleRequirements;
 
         public:
-            std::vector<Vehicle> districtVehicles;
-            std::vector<FireStation> districtFireStations;
-
+            FireStation findFireStation(uint32_t fireStationID);
+            void processVehicleSampleData(const std::string& fileName);
+            void processFireStationSampleData(const std::string& fileName);
+            void processEventSampleData(const std::string &fileName, std::vector<Event> *pendingQueue);
+            void processVehicleRequirementsSampleData(const std::string &fileName);
+            std::vector<Vehicle *> getVehicleRequirements(uint32_t eventID);
```

Comment:
```suggestion
            std::vector<Vehicle *> getVehicleRequirements(uint32_t eventID) const;
```
Not sure if this is correct, just suggesting to think of it.

A `const` member function is one that does not modify any member variables of the class nor return a non-const value.